core/src/test/java/com/graphhopper/routing/ev/EnumEncodedValueTest.java

```diff
@@ -32,4 +40,36 @@ public void testSize() {
         assertEquals(4, 32 - Integer.numberOfLeadingZeros(16 - 1));
         assertEquals(5, 32 - Integer.numberOfLeadingZeros(17 - 1));
     }
+
+    @Test
+    public void serializationAndDeserialization() throws JsonProcessingException {
+        ObjectMapper mapper = new ObjectMapper();
+        mapper.setVisibility(PropertyAccessor.ALL, JsonAutoDetect.Visibility.NONE);
+        mapper.setVisibility(PropertyAccessor.FIELD, JsonAutoDetect.Visibility.ANY);
+
+        List<EncodedValue> encodedValues = new ArrayList<>();
+        // add enum, int, decimal and boolean encoded values
+        encodedValues.add(new EnumEncodedValue<>(RoadClass.KEY, RoadClass.class));
+        encodedValues.add(Lanes.create());
+        encodedValues.add(MaxWidth.create());
+        encodedValues.add(GetOffBike.create());
+
+        String serializedEVs = mapper.writeValueAsString(encodedValues);
+        System.out.println(serializedEVs);
+
+        List<EncodedValue> deserializedEVs = new ArrayList<>();
+        JsonNode jsonNode = mapper.readTree(serializedEVs);
+        for (JsonNode jsonEV : jsonNode) {
+            if (jsonEV.has("enumType"))
+                deserializedEVs.add(mapper.treeToValue(jsonEV, EnumEncodedValue.class));
+            else if (jsonEV.has("factor"))
+                deserializedEVs.add(mapper.treeToValue(jsonEV, DecimalEncodedValueImpl.class));
+            else if (jsonEV.has("type") && "simple_boolean".equals(jsonEV.get("type").asText()))
+                deserializedEVs.add(mapper.treeToValue(jsonEV, SimpleBooleanEncodedValue.class));
+            else
+                deserializedEVs.add(mapper.treeToValue(jsonEV, IntEncodedValueImpl.class));
+        }
+        System.out.println(deserializedEVs);
```

Comment:
Here I serialized and deserialized a list of EncodedValues with different types. It's a rather ugly, because we need to recognize the type of the encoded value for deserialization somehow (e.g. I added an extra field to `SimpleBooleanEncodedValue` just for this...). 

Putting everything into a `List<EncodedValue>` is ok, because in `EncodedValueLookup` we get the type information from the client code like `lookup.getEnumEncodedValue(RoadClass.KEY, RoadClass.class)`.